src/hvm.rs

```diff
@@ -3414,28 +3443,27 @@ pub fn reduce(rt: &mut Runtime, root: u128, mana: u128) -> Result<Ptr, RuntimeEr
           if get_tag(arg0) == NUM && get_tag(arg1) == NUM {
             //eprintln!("op2-num");
             rt.set_mana(rt.get_mana() + Op2NumMana());
-            let op  = get_ext(term);
+            let op  = get_ext(term).try_into().map_err(|_| RuntimeError::InvalidOperation)?;
```

Comment:
Values we get from HVM memory should always be valid, so this should not be a possibility and just panic.

Reply:
Invalid `OP2` cells should be prevented by construction.